Gems/ScriptAutomation/Code/Source/ScriptAutomationScriptBindings.cpp

```diff
@@ -555,22 +579,144 @@ namespace ScriptAutomation
                     resolvedPathB,
                     minDiffFilter);
 
-                AZ_Error("ScriptAutomation", compareOutcome.IsSuccess(),
-                    "Image comparison failed. %s", compareOutcome.GetError().m_errorMessage.c_str());
+                AZ_Error(
+                    "ScriptAutomation", 
+                    compareOutcome.IsSuccess(),
+                    "%s screenshot compare error. Error \"%s\"", 
+                    compareName.c_str(),
+                    compareOutcome.GetError().m_errorMessage.c_str()
+                );
                 if (compareOutcome.IsSuccess())
                 {
-                    AZ_Printf(
-                        "ScriptAutomation",
-                        "Diff score is %.5f from %s and %s.",
-                        compareOutcome.GetValue().m_diffScore,
-                        resolvedPathA.c_str(),
-                        resolvedPathB.c_str());
-                    AZ_Printf(
-                        "ScriptAutomation",
-                        "Filtered diff score is %.5f from %s and %s.",
-                        compareOutcome.GetValue().m_filteredDiffScore,
-                        resolvedPathA.c_str(),
-                        resolvedPathB.c_str());
+                    float diffScore = toleranceLevel.m_filterImperceptibleDiffs
+                        ? compareOutcome.GetValue().m_diffScore
+                        : compareOutcome.GetValue().m_filteredDiffScore;
+
+                    if (diffScore > toleranceLevel.m_threshold)
+                    {
+                        AZ_Error(
+                            "ScriptAutomation", 
+                            false,
+                            "%s screenshot compare failed. Diff score %f exceeds threshold of %f ('%s').",
+                            compareName.c_str(),
+                            diffScore,
+                            toleranceLevel.m_threshold,
+                            toleranceLevel.m_name.c_str()
+                        );
+
+                        // TODO: open image compare app if CVAR is set
+                    }
+                    else
+                    {
+                        AZ_Printf(
+                            "ScriptAutomation",
+                            "%s screenshot compare passed. Diff score is %.5f, threshold of %f ('%s').",
```

Comment:
Should the first format also use `%.5f` precision?

Reply:
I'll update all 4 `%f` to `%.5f`